src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/throw';
+import 'rxjs/add/observable/of';
+import { User } from '../models/user';
+
+export const MOCK_USER = new User();
+MOCK_USER._id = '1';
+MOCK_USER.email = '1';
+MOCK_USER.firstName = 'Foo';
+MOCK_USER.lastName = 'Bar';
+MOCK_USER.password = '1';
+
+/**
+ * The user service.
+ */
+@Injectable()
+export class UserService {
+
+    /**
+     * True if authenticated
+     * @type
+     */
+    private _authenticated = false;
+
+    /**
+     * Authenticate the user
+     *
+     * @param {string} email The user's email address
+     * @param {string} password The user's password
+     * @returns {Observable<User>} The authenticated user observable.
+     */
+    public authenticate(email: string, password: string): Observable<User> {
```

Comment:
methods are public by default - official angular style guide says that it is better to avoid it